source/fundamentals/connection/connection-options.txt

```diff
@@ -6,203 +6,290 @@ Connection Options
 
-   * - **pkFactory**
-     - object
+   * - **readPreferenceTags**
+     - comma-separated key:value pairs (e.g. "dc:ny,rack:1" and "dc:ny)
+       can be specified multiple times, each instance of this key is a
+       separate tag set
      - ``null``
-     - Specifies a primary key factory object that generates custom ``_id``
-       keys.
+     - Specifies the default read preference tags for the client. This option is
+       valid only if the read preference mode is not primary
 
-   * - **promiseLibrary**
-     - object
+       The order of the tag sets in the read preference is the same as the order
+       they are specified in the URI.
+
+   * - **replicaSet**
+     - string
      - ``null``
-     - Specifies the Promise library class the application uses (e.g. Bluebird).
-       This library must be compatible with ES6.
+     - Specifies the name of the replica set to connect to.
 
-   * - **promoteBuffers**
+   * - **retryReads**
      - boolean
-     - ``false``
-     - Specifies whether to promote Binary BSON values to native Node.js
-       ``Buffer`` type data.
+     - ``true``
+     - Enables retryable reads on MongoDB version 3.6 or later.
 
-   * - **promoteLongs**
+   * - **retryWrites**
      - boolean
      - ``true``
-     - Specifies whether to convert ``Long`` values to a number if they fit
-       inside 53 bits of resolution.
+     - Enables retryable writes on MongoDB version 3.6 or later.
+
+   * - **serverSelectionTimeoutMS**
+     - non-negative integer
+     - ``30000``
+     - Specifies the timeout, in milliseconds, to block for server selection before raising an error.
 
-   * - **promoteValues**
+   * - **serverSelectionTryOnce**
      - boolean
      - ``true``
-     - Specifies whether to promote BSON values to Node.js native types when
-       possible. When set to false, it uses wrapper types to present
-       BSON values.
+     - Scan the topology only once after a server selection failure instead of repeatedly until the server selection times out
 
-   * - **raw**
+   * - **socketTimeoutMS**
+     - non-negative integer
+     - ``0``
+     - Specifies the amount of time spent attempting to send or receive on a
+       socket before timing out.
+
+   * - **srvMaxHosts**
+     - non-negative integer
+     - ``0``
+     - Specifies the maximum number of SRV results to randomly select when initially
+       populating the seedlist or, during SRV polling, adding new hosts to the
+       topology.
+
+   * - **srvServiceName**
+     - a valid SRV service name according to `RFC 6335 <https://datatracker.ietf.org/doc/html/rfc6335#section-5.1>`_
+     - ``"mongodb"``
+     - Specifies the service name to use for SRV lookup in initial DNS seedlist discovery.
+
+   * - **ssl**
      - boolean
      - ``false``
-     - Specifies whether to return document results as raw BSON buffers.
+     - ``ssl`` is an alias of "tls"
```

Comment:
Issue: Inconsistent formatting.

Issue: It took me a minute to realize that "tls" was referring to an option. Could this be stated explicitly?

```suggestion
     - The ``ssl`` option is an alias of the ``tls`` option.
```